streamer/input_configuration.py

```diff
@@ -315,10 +315,41 @@ def get_input_args(self) -> List[str]:
   def get_resolution(self) -> bitrate_configuration.VideoResolution:
     return bitrate_configuration.VideoResolution.get_value(self.resolution)
 
+class SinglePeriod(configuration.Base):
+  """An object repersenting one optional video stream and multiple audio and text streams"""
+
+  inputs = configuration.Field(List[Input], required=True).cast()
 
 class InputConfig(configuration.Base):
   """An object representing the entire input config to Shaka Streamer."""
 
-  inputs = configuration.Field(List[Input], required=True).cast()
-  """A list of Input objects, one per input stream."""
+  multiperiod_inputs_list = configuration.Field(List[SinglePeriod]).cast()
+  """A list of SinglePeriod objects"""
+
+  inputs = configuration.Field(List[Input]).cast()
+  """A list of Input objects"""
+
+  def __init__(self, dictionary: Dict[str, Any]):
+    """A constructor to check that either inputs or mutliperiod_inputs_list is provided,
+    and produce a helpful error message in case both or none are provided.
+    
+    We need these checks before passing the input dictionary to the configuration.Base constructor,
+    because it does not check for this 'exclusive or-ing' relationship between fields.
+    """
+
+    assert isinstance(dictionary, dict), """Malformed Input Config File,
+    See some examples at https://github.com/google/shaka-streamer/tree/master/config_files.
+    """
+
+    if dictionary.get('inputs') and dictionary.get('multiperiod_inputs_list'):
```

Comment:
This uses "truthiness", so empty lists will not be matched.  This means this condition would allow:

```yaml
inputs: []
multiperiod_inputs_list:
 - inputs:
   - name: Sintel.2010.4k.mkv
     media_type: video
```

I would prefer to treat this as conflicting fields, so let's use "is not None" here.

Reply:
Sure, that would be better.